### ISSUE TITLE
GROUP BY allowed by CQL parser but doesn't seem to do anything

### ISSUE THREAD

**Issue by smalyshev** [yugabyte/yugabyte-db#3568]
Query like `select * from counters group by type` is allowed by CQL parser, but `GROUP BY` part doesn't seem to have any effect. The docs do not mention `GROUP BY` support, so if it's unsupported, the parser should probably not allow it to avoid confusion.

**Comment by kmuthukk** [comment]
Agreed that we should raise a compile time error rather than silently ignore it. Thx for reporting @smalyshev .
